### PR TITLE
add test class for HttpCoreNIOListener

### PR DESCRIPTION
**Pull request by nuwanw**

### CODE REVIEW

File: modules/transports/core/nhttp/src/test/java/org/apache/synapse/transport/nhttp/NHttpTransportListenerTest.java [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+public class NHttpTransportListenerTest {
+    private static final String HOST = "127.0.0.1";
+    private static final int PORT = 8486;
+    private static HttpCoreNIOListener nHttpListener = new HttpCoreNIOListener();
+    private static TransportInDescription transportInDescription = new TransportInDescription("http");
+
+    @BeforeClass()
+    public static void startListener() throws Exception {
+        Assert.assertFalse("Port already occupied. Can not execute test", TCPUtils.isPortOpen(PORT, HOST));
+        Parameter portParam = new Parameter("port", PORT + "");
```
Comment (chanikag) [inline modules/transports/core/nhttp/src/test/java/org/apache/synapse/transport/nhttp/NHttpTransportListenerTest.java]: Empty string has been concatenated? Is it intentional?

Comment (nuwanw) [reply modules/transports/core/nhttp/src/test/java/org/apache/synapse/transport/nhttp/NHttpTransportListenerTest.java]: Yes. HttpCoreNIOListener accept the port parameter as a string.

File: modules/transports/core/nhttp/src/test/java/org/apache/synapse/transport/nhttp/NHttpTransportListenerTest.java [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+        Assert.assertEquals("Endpoint reference mismatched", ServiceUtils.getServiceEndpoint("myservice", HOST, PORT),
+                endpointReference[0].getAddress());
+
+        endpointReference = nHttpListener.getEPRsForService("t/myservice", HOST);
+        Assert.assertNotNull(endpointReference);
+        Assert.assertEquals("Endpoint reference mismatched", ServiceUtils.getServiceEndpoint("t/myservice", HOST, PORT),
+                endpointReference[0].getAddress());
+
+        endpointReference = nHttpListener.getEPRsForService("myservice.endpoint", HOST);
+        Assert.assertNotNull(endpointReference);
```
Comment (chanikag) [inline modules/transports/core/nhttp/src/test/java/org/apache/synapse/transport/nhttp/NHttpTransportListenerTest.java]: Add an assert message?

Comment (nuwanw) [reply modules/transports/core/nhttp/src/test/java/org/apache/synapse/transport/nhttp/NHttpTransportListenerTest.java]: Done

**Pull request merged**